core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -482,7 +481,15 @@ public void onClosed(Channel c, IOException cause) {
                     taskListener.getLogger().println("Connection terminated");
                 }
                 closeChannel();
-                launcher.afterDisconnect(SlaveComputer.this, taskListener);
+                try {
+                    launcher.afterDisconnect(SlaveComputer.this, taskListener);
+                } catch (Throwable t) {
+                    LogRecord lr = new LogRecord(Level.SEVERE,
+                            "Launcher {0}'s afterDisconnect method propagated an exception when {1}'s connection was closed: {2}");
+                    lr.setThrown(t);
+                    lr.setParameters(new Object[]{launcher, SlaveComputer.this, t.getMessage()});
```

Comment:
@stephenc those that suppress stack traces are ok. (you can print the throwable's getMessage() without the stack trace)
those that suppress throwables entirely are stoopid - and see my comments on the related fix in remoting.
If we don't have a stack trace fixing things becomes very hard, and a message doesn't help anyone.

We should not be trying to help stupid by making things look worse for normal users.
If we want to help stoopid we should educate stoopid on why this is stoopid... (and yes I have done this in the past - its slow progress but the only real way)

Having the message printed twice to me is a sign of a junior developer and doesn't look good. 
I know you don't fit into this category...

I don't care strongly enough to vote -1 based on this - but this style is not followed in the rest of Jenkins or it's plugins so it's really a worthless effort IMHO.